import_3dm/converters/__init__.py

```diff
@@ -38,3 +38,25 @@
     r3d.ObjectType.Mesh : import_render_mesh,
     r3d.ObjectType.Curve : import_curve
 }
+
+
+
+# TODO: Decouple object data creation from object creation
+#       and consolidate object-level conversion.
+
+def convert_object(context, ob, name, layer, rhinomat, scale):
+    """
+    Add a new object with given data, link to
+    collection given by layer
+    """
+
+    data = RHINO_TYPE_TO_IMPORT[ob.Geometry.ObjectType](context, ob, name, scale)
+    data.materials.append(rhinomat)
```

Comment:
Maybe check for data!=None or ObjectType here, or warp it in a try statement? Importing an object of type InstanceReference is not going to import any new mesh data, but create an empty and link it to a collection as an instance. I can make the reference importer return None though..

Reply:
Good catch. I guess `InstanceReference`s are a special case.

Reply:
Ok @lfertig I added a check for null object data, and put a placeholder for handling `InstanceReference`s! Take a look whenever you have a moment